reco_utils/recommender/newsrec/models/naml.py

```diff
@@ -0,0 +1,272 @@
+import numpy as np
+import tensorflow as tf
+import tensorflow.keras as keras
+from tensorflow.keras import layers
+
+
+from reco_utils.recommender.newsrec.models.base_model import BaseModel
+from reco_utils.recommender.newsrec.models.layers import AttLayer2
+
+__all__ = ["NAMLModel"]
+
+class NAMLModel(BaseModel):
+    '''NAML model(Neural News Recommendation with Attentive Multi-View Learning)
+
+    Chuhan Wu, Fangzhao Wu, Mingxiao An, Jianqiang Huang, Yongfeng Huang and Xing Xie,
+    Neural News Recommendation with Attentive Multi-View Learning, IJCAI 2019
+
+    Attributes:
+        word2vec_embedding (numpy.array): Pretrained word embedding matrix.
+        hparam (obj): Global hyper-parameters.
+    '''
+    
+    def __init__(self, hparams, iterator_creator_train, iterator_creator_test):
+        """Initialization steps for NAML.
+        Compared with the BaseModel, NAML need word embedding.
+        After creating word embedding matrix, BaseModel's __init__ method will be called.
+        
+        Args:
+            hparams (obj): Global hyper-parameters. Some key setttings such as filter_num are there.
+            iterator_creator_train(obj): NAML data loader class for train data.
+            iterator_creator_test(obj): NAML data loader class for test and validation data
+        """
+        
+        self.word2vec_embedding = self._init_embedding(hparams.wordEmb_file)
+        self.hparam = hparams
+        
+        super().__init__(hparams, iterator_creator_train, iterator_creator_test)
+
+    def _init_embedding(self, file_path):
+        """Load pre-trained embeddings as a constant tensor.
+        
+        Args:
+            file_path (str): the pre-trained embeddings filename.
+
+        Returns:
+            np.array: A constant numpy array.
+        """
+        return np.load(file_path).astype(np.float32)
+
+
+    def _build_graph(self):
+        """Build NAML model and scorer.
+
+        Returns:
+            obj: a model used to train.
+            obj: a model used to evaluate and inference.
+        """
+
+        model, scorer = self._build_naml()
+        return model, scorer
+
+    def _build_userencoder(self, newsencoder):
+        '''The main function to create user encoder of NAML.
```

Comment:
I've noticed that in some files you added `'''` and in others `"""`. Normally what we do is to use black, that automatically formats everything when you save. I can do it for you if you want. 

In case you want to add it yourself, I can tell you very quickly how to activate it in VSCode. You just need to have a conda environment with black installed and then add this line to your configuration https://github.com/miguelgfierro/codebase/blob/master/minsc/vscode_settings.json#L7

If you or anyone in your team want to know how to activate this, I'll be happy to jump on a quick call to help you do this very quickly

Reply:
Thanks, I will try add it.